plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/BaseMongoDistributedQueries.java

```diff
@@ -78,10 +78,9 @@ protected TestTable createTableWithDefaultColumns()
     public void testColumnName(String columnName)
     {
         if (columnName.equals("a.dot")) {
-            // TODO (https://github.com/trinodb/trino/issues/3460)
-            assertThatThrownBy(() -> super.testColumnName(columnName))
-                    .hasStackTraceContaining("TableWriterOperator") // during INSERT
-                    .hasMessage("Invalid BSON field name a.dot");
+            assertThatThrownBy(() -> super.testColumnName(columnName)).hasCauseInstanceOf(RuntimeException.class)
+                    .hasStackTraceContaining("java.lang.IllegalArgumentException") // during INSERT
+                    .hasMessage("Column name must not contain '$' or '.' for INSERT: " + columnName);
```

Comment:
```java
        if (columnName.equals("a.dot")) {
            assertThatThrownBy(() -> super.testColumnName(columnName))
                    .isInstanceOf(RuntimeException.class)
                    .hasMessage("Column name must not contain '$' or '.' for INSERT: " + columnName);
            throw new SkipException("Insert would fail");
        }
```
```